src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -696,6 +696,52 @@ public function getQueryStringNormalizationData()
             // Ignore pairs with empty key, even if there was a value, e.g. "=value", as such nameless values cannot be retrieved anyway.
             // PHP also does not include them when building _GET.
             array('foo=bar&=a=b&=x=y', 'foo=bar', 'removes params with empty key'),
+
+            // Also reorder nested query string keys
+            array('foo[]=Z&foo[]=A', 'foo%5B%5D=A&foo%5B%5D=Z', 'reorders values'),
+            array('foo[Z]=B&foo[A]=B', 'foo%5BA%5D=B&foo%5BZ%5D=B', 'reorders keys'),
+        );
+    }
+
+    /**
+     * @dataProvider getQueryStringNormalizationDataForPhp
+     */
+    public function testGetQueryStringForPhp($query, $expectedQuery, $msg)
+    {
+        $request = new Request();
+
+        $request->server->set('QUERY_STRING', $query);
+        $this->assertSame($expectedQuery, $request->getQueryStringForPhp(), $msg);
+    }
+
+    public function getQueryStringNormalizationDataForPhp()
```

Comment:
Could we add some tests with UTF-8 chars in the query string? You can see those in URLs like `https://github.com/symfony/symfony/issues?utf8=✓`

Reply:
added